pkg/webhook/merge_on_review_approval_and_status_checks.go

```diff
@@ -40,27 +40,41 @@ func (h *autoMerger) HandleEvent(w http.ResponseWriter, payload interface{}, ghC
 		return h.handlePullRequestEvent(w, event, ghClientFunc)
 	case *github.StatusEvent:
 		return h.handleStatusEvent(w, event, ghClientFunc)
+	case *github.PullRequestReviewEvent:
+		return h.handlePullRequestReviewEvent(w, event, ghClientFunc)
 	default:
 		return nil
 	}
 }
 
+func (h *autoMerger) handlePullRequestReviewEvent(w http.ResponseWriter, event *github.PullRequestReviewEvent, ghClientFunc GitHubIntegrationsClientFunc) error {
+	if strings.ToLower(event.Review.GetState()) != approvedReviewState {
+		return nil
+	}
+
+	return h.mergePRFromPullRequestEvent(event.Installation.GetID(), event.Repo, event.PullRequest, ghClientFunc)
+}
+
 func (h *autoMerger) handlePullRequestEvent(w http.ResponseWriter, event *github.PullRequestEvent, ghClientFunc GitHubIntegrationsClientFunc) error {
 	if strings.ToLower(event.GetAction()) != labeledEvent {
```

Comment:
There should be a check for the `approved` label too... Ideally it would check that all PR reviews pass but GitHub integrations can't access reviews API endpoints (full list of accessible endoints [here](https://developer.github.com/early-access/integrations/available-endpoints/)).